test/Sentry.Tests/Internals/Http/CachingTransportTests.cs

```diff
@@ -61,51 +61,75 @@ public async Task ShouldNotLogOperationCanceledExceptionWhenIsCancellationReques
             Debug = true
         };
 
+        var capturingSync = new ManualResetEventSlim();
```

Comment:
Test was flaky because it relied on stuff happening within `100ms`.

Now the test synchronizes both threads to avoid any Thread.Sleep and flakiness